web/src/list_widget.js

```diff
@@ -129,6 +131,37 @@ function is_scroll_position_for_render(scroll_container) {
     );
 }
 
+export function toggle_empty_list_message($container, filter_value) {
+    let empty_list_message = $container.data("empty");
+
+    const empty_search_results_message = $container.data("search-results-empty");
+    if (filter_value && empty_search_results_message) {
+        empty_list_message = empty_search_results_message;
+    }
+
+    if (!empty_list_message || $container.children().length) {
+        return;
+    }
+
+    // Get the number of columns to span the message across all columns
+    // if the container is a table
+    // This helps in styling the message consistently for different containers
```

Comment:
This small chunk of code seems really easy to extract into a function called `get_column_count_for_table($table)` and call it with `const column_count = get_column_count_for_table($container)`.